### ISSUE TITLE
Get rid of intro text.

### ISSUE THREAD

**Issue by ftyers** [goavki/apertium-html-tools#269]
Get rid of the following:

```
Choose two languages and translate your text using Apertium's translation platform!
Choose a language and analyse your text using Apertium's morphological analysers.
Choose a language and generate surface forms from your text using Apertium's morphological generators.
```

They take a lot of space and are superfluous.

**Comment by jonorthwash** [comment]
Maybe it would be good to have an explanation somewhere, though?  If you just come across the site randomly, you might not know what it's for.

Perhaps replace with
```
Translate text from one language to another using Apertium.
Analyse text using Apertium's morphological analysers.
Generate text from morphological forms using Apertium's morphological analysers.
```
?

**Comment by sushain97** [comment]
Yeah, that sounds potentially better.

**Comment by jonorthwash** [comment]
Links to documentation, especially for how to use the generator, might be good?  Or should we figure out how to put some example form for the transducer in question?

**Comment by ftyers** [comment]
No no no :D ... please get rid of them. Another option would be to have a question mark icon somewhere with a tooltip, but they are _massive_ waaaaay too big.

**Comment by unhammer** [comment]
In http://jorgal.uit.no/index.sme.html?dir=sme-nob#translation we use the placeholder attribute on the input box because of feedback from people who thought the text was a waste of space. Also removed  some other stuff from the main page as well as the html translation page (well, apertium.org should keep the lang selectors ;)).

**Comment by ftyers** [comment]
I've commented it out in  ac52067 and updated the apertium website, now it looks much better.

**Comment by sushain97** [comment]
I still don't agree with completing removing the text. They are redundant to us but we've used the interface hundreds of times. I'm going to move your commit to a branch until either a) we come to a consensus to remove it or b) we go with @jonorthwash or @unhammer's suggestions (shorter text or placeholder) across the modes which I'm fine with. Feel free to use the branch for the Apertium website or comment it out locally but there's no need to have a half-baked 'solution' in master.

**Comment by ftyers** [comment]
@sushain97 ... it's obvious because that's how Google works, how Yandex works and how most system works.

* Google (translate.google.com): tiny text underneath box ... good design
* Yandex (https://translate.yandex.ru/): text inside entry box that disappears... good design

... 

* Apertium: massive thing at the top of the page taking up a tonne of space which makes parts of the UI go off the screen ... bad design.

![captura de 2018-02-20 21-58-28](https://user-images.githubusercontent.com/449545/36448846-334cbe10-1689-11e8-9980-46d4710ac80a.png)

Remove the massive redundant thing and everything fits nicely:

![captura de 2018-02-20 21-59-30](https://user-images.githubusercontent.com/449545/36448896-604a0c42-1689-11e8-9539-d96a65a0c4a1.png)

And in addition the interface is as obvious how to use as ever. I've fixed it on www.apertium.org and I guess it will do for now.

**Comment by sushain97** [comment]
The translation interface isn't the only thing that's relevant here. There
are two others which aren't as clear and consistency is important.

Whether it forces parts of the UI out of the viewport is completely a
function of your resolution. I've never experienced that problem on any
machine I've used to access apertium.org but my personal experience isn't
relevant either.

I'm fine with Unhammer's solution since I do agree on some viewports the
text can be large. Regardless, my real problem was a commit to master that
doesn't address the full problem and leaves dead code and strings lying
around. Ideally, the build would fail with dead strings... Sounds like a
fun little sed.

On Feb 20, 2018 3:02 PM, "Francis Tyers" <notifications@github.com> wrote:

@sushain97 <https://github.com/sushain97> ... it's obvious because that's
how Google works, how Yandex works and how most system works.

   - Google (translate.google.com): tiny text underneath box ... good design
   - Yandex (https://translate.yandex.ru/): text inside entry box that
   disappears... good design

...

   - Apertium: massive thing at the top of the page taking up a tonne of
   space which makes parts of the UI go off the screen ... bad design.

[image: captura de 2018-02-20 21-58-28]
<https://user-images.githubusercontent.com/449545/36448846-334cbe10-1689-11e8-9980-46d4710ac80a.png>

Remove the massive redundant thing and everything fits nicely:

[image: captura de 2018-02-20 21-59-30]
<https://user-images.githubusercontent.com/449545/36448896-604a0c42-1689-11e8-9539-d96a65a0c4a1.png>

And in addition the interface is as obvious how to use as ever. I've fixed
it on www.apertium.org and I guess it will do for now.

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/goavki/apertium-html-tools/issues/269#issuecomment-367117944>,
or mute the thread

**Comment by ftyers** [comment]
It's not really a change I wanted to make, I figured someone else with more experience with the code might want to do it. But in any case I did it because after the _n_th time of the text appearing I had had enough. I shouldn't need to learn git in order to fix something obviously wrong with www.apertium.org.

**Comment by sushain97** [comment]
Is it not possible to edit `index.html.in` locally? There's no need to involve Git and as long as no one edits that line, you won't have any trouble. I doubt that line has been edited for months (years?).

**Comment by ftyers** [comment]
That's what I've done (again). But I keep commenting out locally and it keeps coming back, and it's not something I really want to debug, I just want it fixed. Perhaps the best idea is to fork the repo and make the Apertium installations use that ?

**Comment by sushain97** [comment]
I don't know why your local changes would keep disappearing unless you're doing more than a `git pull`. Git makes it very hard to lose local changes when updating. What command are you using? You can also `git commit` your comment without pushing it and then use `git pull --rebase` and that should ensure it doesn't get erased.

Forking is always possible but merging changes from origin into a fork is IMO much more tricky than branching. Also, contributing to Apertium requires learning SVN so I don't see why contributing to html-tools (and by consequence modifying apertium.org) having Git competency as a prerequisite is any less justified when all the active devs prefer Git.

**Comment by ftyers** [comment]
I don't know why either. And I don't keep a track of all the git commands I copy and paste into the terminal each time the thing breaks. To be able to modify apertium.org you need an account on the host, xixona which is not easy to get. As far as I know it's also not easy to change where apertium.org points. So we are kind of stuck with it.

**Comment by jonorthwash** [comment]
It sounds like everyone is okay with @unhammer's and Yandex's solution.  Let's go with that?

**Comment by ftyers** [comment]
@jonorthwash that's fine let me know when you've done it!

**Comment by jonorthwash** [comment]
If we go that way (the placeholder text that disappears), we should probably reword it to something like this:
```
Enter text to translate and select source and destination languages.
Enter text for morphological analysis.
Enter morphological forms that you would like generated into text.
```

**Comment by sushain97** [comment]
aa046dd implements the placeholder version of the texts. Whether to reword them or not is a bit more involved since we'll have to delete all the existing localizations. There's no apparatus for per localization fallbacks so that we could use the existing versions if they exist and the old otherwise. If we just leave the old ones here, we'll have misleading #s on completion and will _never_ remember to fix them. @jonorthwash do you think it's worth it to reword?

Either way, unless there are any objections, I think this particular issue is safe to close. Rewording the placeholders belongs in another issue.

**Comment by jonorthwash** [comment]
What about changing the name of the string (i.e., adding a new string for this and removing the old one from the English localisation)?

**Comment by sushain97** [comment]
Well, that just means we have a dead string. What I meant by

> There's no apparatus for per localization fallbacks so that we could use the existing versions if they exist and the old otherwise.

is that I don't think the two string solution would work. Sure, we could modify the code to support it... but seems like just letting people localize three sentences again instead of trying to build legacy-style support is OK here.

**Comment by sushain97** [comment]
Actually, in light of https://github.com/goavki/apertium-html-tools/issues/198, I'm really leaning towards just ditching at least the existing translation string.

**Comment by jonorthwash** [comment]
> seems like just letting people localize three sentences again instead of trying to build legacy-style support is OK here.

Yeah, I think it's not a real problem.  Keeping localisations up to date is going to be a constant struggle anyway.  Dwelling o the level of bit rot we might cause in that domain is probably not the best decision-making strategy.
